Audio device selection

default0 opened the pull request:
Allows changing the audio device in the settings menu.
Depends on https://github.com/ppy/osu-framework/pull/436 and https://github.com/ppy/osu-framework/pull/437.

default0 committed:
Update OsuDropDownMenu for Framework-Changes

Since the DropDownMenu in the framework has changed it was necessary
to update the GetDropDownItems override of OsuDropDownMenu to
accomodate the new structure of the framework.

default0 committed:
Rework OptionDropdowns to be more versatile

The existing OptionDropdown only supported enums and was thus renamed
to OptionEnumDropDown. A new OptionDropdown has been created in its
place to allow binding to arbitrary values, with a set of user-provided
items.

default0 committed:
Allow audio device selection in settings

Tom94 commented on osu.Game/Overlays/Options/Sections/Audio/AudioDevicesOptions.cs:
```diff
@@ -1,18 +1,41 @@
+            base.LoadComplete();
+
+            var deviceItems = new List<KeyValuePair<string, string>>();
+            deviceItems.Add(new KeyValuePair<string, string>("Standard", ""));
```
In english windows the default device is called "default", not "standard". Let's aim for consistency.

default0 committed:
Rename Standard device option to Default

Changed the name of the option for using the default audio device
to "Default" for consistency with english windows. Also changed
the "" value to string.Empty for consistency with the Frameworks
treatment of this special value.

peppy committed:
Merge branch 'master' into audio-device-selection

# Conflicts:
#	osu-framework
#	osu.Game/Overlays/Options/Sections/Audio/AudioDevicesOptions.cs

peppy commented:
Merging this, but we still need to refresh the dropdown when device availability changes.
